### PR TITLE
WT-7712 commit and durable timestamps should be disallowed at stable timestamp

### PR DESCRIPTION
**Pull request by keithbostic**
@kommiharibabu, @sauclovian-wt, would you please do the review? I tried to be careful in the test suite to be sure my changes didn't alter the intent of the test, but obviously that's the most interesting question for the test suite.

### CODE REVIEW

**Comment by keithbostic** [comment]
This change does not yet apply to MongoDB builds and can be independently committed without changing MongoDB server code. As a data point for review, here's a patch run to see what breaks:
https://evergreen.mongodb.com/version/621bc7bd2a60ed43e2df5c7c?redirect_spruce_users=true

**Review by kommiharibabu — APPROVED** [review]
LGTM

File: dist/api_data.py [hunk 1]
```diff
@@ -1776,18 +1775,20 @@ def __ge__(self, other):
...
-        commit of a prepared transaction, and otherwise not permitted. The
-        supplied value must not be older than the commit timestamp set for the
-        current transaction. The value must also be newer than the current
-        oldest and stable timestamps. See @ref timestamp_prepare'''),
+        set the durable timestamp for the current transaction. Required for
+        the commit of a prepared transaction, and otherwise not permitted. Can
+        only be set after the transaction has been prepared and a commit
+        timestamp has been set. The value must not be older than
+        the commit timestamp. See @ref timestamp_prepare'''),
     Config('operation_timeout_ms', '0', r'''
```
Comment (sauclovian-wt) [inline dist/api_data.py:1792]: This should still say that it must be newer than the current stable timestamp. (And oldest, but that's implied so doesn't matter one way or the other whether it's explicitly stated.)

File: dist/api_data.py [hunk 2]
```diff
@@ -1806,35 +1807,35 @@ def __ge__(self, other):
...
-        be set once the current transaction has been prepared, and a commit
-        timestamp has been set. The supplied value must not be older than the
-        commit timestamp. The value must also be newer than the current
-        oldest and stable timestamps. See @ref timestamp_prepare'''),
+        set the durable timestamp for the current transaction. Required for
+        the commit of a prepared transaction, and otherwise not permitted. Can
+        only be set after the transaction has been prepared and a commit
+        timestamp has been set. The value must not be older than
+        the commit timestamp. See @ref timestamp_prepare'''),
     Config('prepare_timestamp', '', r'''
```
Comment (sauclovian-wt) [inline dist/api_data.py:1830]: Likewise here.

File: src/docs/timestamp-txn.dox [hunk 3]
```diff
@@ -65,8 +65,8 @@ transaction commits, using WT_SESSION::commit_transaction:
 
 | Timestamp | Constraint | Description |
 |-----------|------------|-------------|
-| commit_timestamp | >= stable and >= prepare and >= any system read timestamp | the transaction's commit timestamp, see @ref timestamp_txn_api_commit_timestamp for details |
-| durable_timestamp | >= commit and > stable | the transaction's durable timestamp, only applicable to prepared transactions, see @ref timestamp_prepare for details |
+| commit_timestamp | > stable and >= prepare and >= any system read timestamp | the transaction's commit timestamp, see @ref timestamp_txn_api_commit_timestamp for details |
+| durable_timestamp | >= commit | the transaction's durable timestamp, only applicable to prepared transactions, see @ref timestamp_prepare for details |
```
Comment (sauclovian-wt) [inline src/docs/timestamp-txn.dox:70]: (and here)

File: src/docs/timestamp-txn.dox [hunk 4]
```diff
@@ -84,8 +84,8 @@ points in the transaction's lifetime, using WT_SESSION::timestamp_transaction:
...
 | prepare_timestamp | > stable and >= any system read timestamp | the transaction's prepare timestamp, see @ref timestamp_prepare for details |
```
Comment (sauclovian-wt) [inline src/docs/timestamp-txn.dox:89]: (and here)

**Review by sauclovian-wt — APPROVED** [review]
One thing, otherwise looks good. (I don't immediately see why the hs24 change is needed, but it also appears harmless, so no big deal.)

File: dist/api_data.py (hunk 1 above)
Comment (keithbostic) [reply dist/api_data.py:1789]: Fixed in 493d1dd.

File: dist/api_data.py (hunk 2 above)
Comment (keithbostic) [reply dist/api_data.py:1826]: Fixed in 493d1dd.

File: src/docs/timestamp-txn.dox (hunk 3 above)
Comment (keithbostic) [reply src/docs/timestamp-txn.dox:70]: Fixed in 493d1dd.

File: src/docs/timestamp-txn.dox (hunk 4 above)
Comment (keithbostic) [reply src/docs/timestamp-txn.dox:89]: Fixed in 493d1dd.

**Pull request merged**